### DISCUSSION TITLE
How to supress "data.map" is not a function error

### DISCUSSION THREAD

**Discussion by Joe1080** [Budibase/budibase#8459] [Help]
I use API queries in my pages, I redirect the user to login if the JWT token isn't valid but the above-mentioned error still appears. Is there any way to not show errors like this? I think it's from the transformer I use in the API queries which fails when the token is bad or doesn't exist. **Screenshot in replies.

**Comment by Joe1080** [comment]
![Screenshot (104)](https://user-images.githubusercontent.com/84574736/199222842-f1111aba-2ec8-417e-938f-b3600304e036.png)

**Comment by andz-bb** [comment]
Hey @Edwin-Joe 

you could try something like this in your transformer before the data.map:

```
if (Object.keys(data).length === 0) {
	return {};
}

return data.map(.......
```

This will check if the data object is empty and will skip running the map on it if it is. Hope that helps!
